loopd/swapclient_server.go

```diff
@@ -433,11 +438,16 @@ func (s *swapClientServer) LoopIn(ctx context.Context,
 	np2wshAddress := swapInfo.HtlcAddressNP2WSH.String()
 
 	response := &looprpc.SwapResponse{
-		Id:                swapInfo.SwapHash.String(),
-		IdBytes:           swapInfo.SwapHash[:],
-		HtlcAddress:       np2wshAddress,
-		HtlcAddressP2Wsh:  p2wshAddress,
-		HtlcAddressNp2Wsh: np2wshAddress,
+		Id:               swapInfo.SwapHash.String(),
+		IdBytes:          swapInfo.SwapHash[:],
+		HtlcAddressP2Wsh: p2wshAddress,
+	}
+
+	if in.ExternalHtlc {
+		response.HtlcAddress = np2wshAddress
+		response.HtlcAddressNp2Wsh = np2wshAddress
+	} else {
+		response.HtlcAddress = p2wshAddress
```

Comment:
Code duplication from above?

Reply:
I may be missing something here but don't understand what you mean.

Reply:
Populating the three address fields for loop in is code that can be extracted to a function and used here and in `marshallSwap`

Reply:
Discussed offline. Not so easy because `SwapInfo` and `LoopInSwapInfo` are different types